### DISCUSSION TITLE
Got a 'Expected 0 arguments but got 1' when using useContractWrite

### DISCUSSION THREAD

**Discussion by AlphaDio** [wevm/wagmi#3079] [Q&A]
I followed the instructions on the useContractWrite page, especially when it comes to preparing the contract write, to use the write function with dynamic arguments.

But lo, of course the write function doesn't seem to play well with typescript and tells me it expected 0 args, when I expected the amount of arguments to be inferred.

Should I just use viem and not bother with the wagmi hooks?

**Comment by tmm** [comment]
Have some code to look at? Hard to tell what's going on without it.
